libjolie/src/main/java/jolie/lang/parse/ast/InputPortInfo.java

```diff
@@ -84,4 +87,23 @@ public OLSyntaxNode location() {
 	public void accept( OLVisitor visitor ) {
 		visitor.visit( this );
 	}
+
+	/**
+	 * @return an abstract node with defines protocol id this port is considering
+	 */
+	public OLSyntaxNode protocolId() {
+		OLSyntaxNode p = this.protocol;
+		if( p instanceof SumExpressionNode ) {
+			// a non optimized protocol node
+			OLSyntaxNode prodVal = ((SumExpressionNode) protocol).operands().iterator().next().value();
+			p = ((ProductExpressionNode) prodVal).operands().iterator().next().value();
+		}
+
+		if( p instanceof InlineTreeExpressionNode ) {
+			return ((InlineTreeExpressionNode) p)
+				.rootExpression();
+		} else {
+			return p;
+		}
+	}
```

Comment:
I am not sure that, in order to bring back protocolId() in this commit, I should modify OLParser to separately parsing protocolId and protocol configuration as we had before this commit? Is this method make sense for you?

Reply:
I think what you had was mostly fine. What I meant was simply that client code interested in getting the protocol id shouldn't get an entire OLSyntaxNode and then use `toString()`. That logic should be put inside of the method `protocolId()`. So you should have both your `protocol()` and the `protocolId()` methods. Does that make sense?